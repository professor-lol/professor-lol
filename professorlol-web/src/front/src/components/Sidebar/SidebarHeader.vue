<template>
  <div v-if="hasSlotDefault" class="sidebar-header">
    <slot></slot>
  </div>
</template>
<script>

export default {
  name: 'sidebar-header',
  computed: {
    hasSlotDefault() {
      return !!this.$slots.default;
    }
  }
}
</script>
