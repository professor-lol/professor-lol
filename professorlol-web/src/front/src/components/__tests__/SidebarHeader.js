import { mount } from '@vue/test-utils'
import SidebarHeader from "../Sidebar/SidebarHeader";

describe("SidebarHeader.vue", () => {
  // Inspect the raw component options
  it('has hasSlotDefault computed property', () => {
    expect(typeof SidebarHeader.computed.hasSlotDefault).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarHeader, { slots: { default: 'test'}})
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('sidebar-header')
  })
});
