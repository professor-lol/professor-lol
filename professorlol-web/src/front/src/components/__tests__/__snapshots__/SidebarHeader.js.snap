// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarHeader.vue renders correctly 1`] = `
<div
  class="sidebar-header"
>
  test
</div>
`;
